addons/stock/models/stock_quant.py

```diff
@@ -43,8 +43,8 @@ def _domain_product_id(self):
         if not self._is_inventory_mode():
             return
         domain = [('type', '=', 'product')]
-        if self.env.context.get('product_tmpl_id'):
-            domain = expression.AND([domain, [('product_tmpl_id', '=', self.env.context['product_tmpl_id'])]])
+        if self.env.context.get('product_tmpl_ids'):
+            domain = expression.AND([domain, [('product_tmpl_id', 'in', self.env.context['product_tmpl_ids'])]])
         return domain
```

Comment:
I think you could had this condition to manage the case where your kit is constitute with only one product:
```suggestion
        if self.env.context.get('product_tmpl_ids'):
            domain = expression.AND([domain, [('product_tmpl_id', 'in', self.env.context['product_tmpl_ids'])]])
        elif self.env.context.get('default_product_id'):
            domain = expression.AND([domain, [('id', '=', self.env.context['default_product_id'])]])
        return domain
```
But as this case seems a little dubious, I don't really mind.